config/config.go

```diff
@@ -78,6 +81,9 @@ type Config struct {
 	envsLock                sync.RWMutex // protects the envs map below
 	envs                    map[string]string
 	envPrefix               string // prefix for environment variables
+	atomicVars              map[string]any
+	atomicVarsMisuses       map[string]string
+	atomicVarsLock          sync.RWMutex // used to protect both atomicVars maps
```

Comment:
**Minor:** since all these fields are part of the hot-reloadable feature we could use `hotReloadableConfigLock` instead of `atomicVarsLock`

Reply:
They are used at different times and to protect different data so I'm keen to keep both actually to minimize lock contention. It seems like a small price to pay.